src/Dialog.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react'
+import {Dialog as ReachDialog} from '@reach/dialog'
+import raw from 'raw.macro'
+import styled, {createGlobalStyle} from 'styled-components'
+import PropTypes from 'prop-types'
+import {space, color} from 'styled-system'
+import systemPropTypes from '@styled-system/prop-types'
+import {X} from '@primer/octicons-react'
+import StyledOcticon from './StyledOcticon'
+import {LAYOUT} from './constants'
+import theme from './theme'
+import Text from './Text'
+import Flex from './Flex'
+
+const reachStyles = raw('@reach/dialog/styles.css')
+
+const ReachGlobalStyle = createGlobalStyle`
+  ${reachStyles}
+
+  [data-reach-dialog-overlay] {
+    z-index: 1000002; /* Higher than the Dropdown and Tooltip */
+  }
+`
+
+export const Dialog = styled(ReachDialog)`
+  box-shadow: 0px 4px 32px rgba(0, 0, 0, 0.35);
+  border-radius: 4px;
+  padding: 0 !important;
+
+  @media screen and (max-width: 750px) {
+    width: 100vw !important;
+    margin: 0 !important;
+    border-radius: 0;
+    height: 100vh;
+  }
+
+  ${LAYOUT}
+  ${space}
+  ${color}
+`
+
+const UnstyledButton = styled(Flex).attrs({
+  as: 'button'
+})`
+  background: none;
+  border: none;
+  padding: none;
```

Comment:
Whoops, missed this the first time around! `padding: none` isn't valid CSS, this should be `padding: 0`
```suggestion
  padding: 0;
```